engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ArrowOperationPlus.java

```diff
@@ -0,0 +1,75 @@
+package org.enso.interpreter.arrow.runtime;
+
+import com.oracle.truffle.api.interop.ArityException;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.InvalidArrayIndexException;
+import com.oracle.truffle.api.interop.TruffleObject;
+import com.oracle.truffle.api.interop.UnknownIdentifierException;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.interop.UnsupportedTypeException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+
+@ExportLibrary(InteropLibrary.class)
+public final class ArrowOperationPlus implements TruffleObject {
+  private final ArrowFixedSizeArrayFactory factory;
+
+  public ArrowOperationPlus(ArrowFixedSizeArrayFactory factory) {
+    this.factory = factory;
+  }
+
+  @ExportMessage
+  boolean isExecutable() {
+    return true;
+  }
+
+  @ExportMessage
+  Object execute(
+      Object[] args,
+      @CachedLibrary(limit = "10") InteropLibrary iop,
```

Comment:
Why so many dispatched interop libraries? Isn't it enough to provide just the `iop` one? All of these will be polymorphic at the end, no? If you want the other ones to be monomorphic, you should declare them without `limit` and only with `value` expression. No?

Reply:
The behavior is faster when each `InteropLibrary` is specialized to a single type of receiver. But yes, I agree this might not be ideal (yet).

> you should declare them without limit and only with value expression

Then I would need to specify limit in `@ExportMessage`. I have never done that yet.